### PR TITLE
Fix overriding of fixtures with parametrization.

### LINKED ISSUE #979: Regression: duplicate 'fixt' when overriding parametrized fixture with parametrize
This test:

```python
import pytest

@pytest.fixture(params=[1, 2])
def fixt(request):
    return request.param

@pytest.mark.parametrize('fixt, val', [(3, 'x'), (4, 'x')])
def test_foo(fixt, val):
    pass
```

works with 2.7.2 but fails since #926 / 41cef6f5f2a20b918bb9b9204ca6c91b47442d15:

```
.venv/lib/python3.4/site-packages/pluggy.py:724: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.venv/lib/python3.4/site-packages/pluggy.py:338: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.venv/lib/python3.4/site-packages/pluggy.py:333: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
.venv/lib/python3.4/site-packages/pluggy.py:595: in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
.venv/lib/python3.4/site-packages/pluggy.py:249: in _wrapped_call
    wrap_controller.send(call_outcome)
_pytest/python.py:285: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
_pytest/python.py:442: in _genfunctions
    self.ihook.pytest_generate_tests(metafunc=metafunc)
.venv/lib/python3.4/site-packages/pluggy.py:724: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.venv/lib/python3.4/site-packages/pluggy.py:338: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.venv/lib/python3.4/site-packages/pluggy.py:333: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
.venv/lib/python3.4/site-packages/pluggy.py:596: in execute
    res = hook_impl.function(*args)
_pytest/python.py:195: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
_pytest/python.py:967: in parametrize
    param_index)
_pytest/python.py:819: in setmulti
    self._checkargnotcontained(arg)
_pytest/python.py:802: in _checkargnotcontained
    raise ValueError("duplicate %r" %(arg,))
E   ValueError: duplicate 'fixt'
```

It only seems to fail if:

* The fixture is parametrized
* It's overridden by a parametrized argument
* There is at least one other parametrized argument.

I agree overriding a fixture this way is a bit weird, but I think it should still work.

/cc @untitaker @bubenkoff 

by the way - thanks to git bisect this was easy to find out:

```
git bisect start
git bisect bad master
git bisect good 2.7.2
git bisect run ./.venv/bin/py.test test_foo.py
```

### PR DESCRIPTION
**Pull request by The-Compiler**
Fixes #979, see #926.

/cc @untitaker

### CODE REVIEW

File: testing/python/fixture.py [hunk 1]
```diff
@@ -1614,6 +1614,22 @@ def test_issue(foo, foobar):
...
+    @pytest.mark.parametrize('param_args', ["'fixt, val'", "'fixt,val'", "['fixt', 'val']", "('fixt', 'val')"])
+    def test_override_parametrized_fixture_issue_979(self, testdir, param_args):
+        testdir.makepyfile("""
+            import pytest
+
+            @pytest.fixture(params=[1, 2])
+            def fixt(request):
+                return request.param
+
+            @pytest.mark.parametrize(%s, [(3, 'x'), (4, 'x')])
```
Comment (bubenkoff) [inline testing/python/fixture.py]: pls use .format for string formatting

File: testing/python/fixture.py [hunk 2]
```diff
@@ -1614,6 +1614,22 @@ def test_issue(foo, foobar):
         reprec = testdir.inline_run()
         reprec.assertoutcome(passed=9)
 
+    @pytest.mark.parametrize('param_args', ["'fixt, val'", "'fixt,val'", "['fixt', 'val']", "('fixt', 'val')"])
+    def test_override_parametrized_fixture_issue_979(self, testdir, param_args):
+        testdir.makepyfile("""
```
Comment (bubenkoff) [inline testing/python/fixture.py]: pls document what exactly you're testing

File: testing/python/fixture.py (hunk 1 above)
Comment (The-Compiler) [inline testing/python/fixture.py]: `.format` isn't used anywhere in this file, and `%`-formatting is used 23 times (which is why I used that) - do you really want to introduce that inconsistency?

Comment (bubenkoff) [inline testing/python/fixture.py]: valid point, agreed
ignore this

**Comment by bubenkoff** [comment]
thanks!

**Pull request merged**